src/backend/api/meals.js

```diff
@@ -2,14 +2,213 @@ const express = require("express");
+    query = query.whereRaw("title LIKE ?", [`%${request.query.title}%`]);
+  }
+
+  if (
+    "dateAfter" in request.query &&
+    typeof request.query.dateAfter === "string"
+  ) {
+    query = query.whereRaw("`when` > ?", [request.query.dateAfter]);
+  }
+
+  if (
+    "dateBefore" in request.query &&
+    typeof request.query.dateBefore === "string"
+  ) {
+    query = query.whereRaw("`when` > ?", [request.query.dateBefore]);
+  }
+
+  if ("limit" in request.query && typeof request.query.limit === "string") {
+    query = query.limit(`${request.query.limit}`);
+  }
+
+  if (request.query.sort_key && request.query.sort_dir) {
+    const sortAllowed = ["when", "max_reservations", "price"];
+    const orderBy = request.query.sort_key.toString();
+    let queryIncludesAllowedSort = sortAllowed.some((element) =>
+      orderBy.includes(element)
+    );
+    if (queryIncludesAllowedSort) {
+      query = query.orderByRaw(`${orderBy} ${request.query.sort_dir}`);
+    } else {
+      response
+        .status(404)
+        .send("Bad request, only when, max_reservations, price allowed");
+    }
+  }
+
+  if (request.query.sort_key && !request.query.sort_dir) {
+    const sortAllowed = ["when", "max_reservations", "price"];
+    const orderBy = request.query.sort_key.toString();
+    let queryIncludesAllowedSort = sortAllowed.some((element) =>
+      orderBy.includes(element)
+    );
+    if (queryIncludesAllowedSort) {
+      query = query.orderByRaw(orderBy);
+    } else {
+      response
+        .status(404)
+        .send("Bad request, only when, max_reservations, price allowed");
+    }
+  }
+  try {
+    const titles = await query;
+    response.json(titles);
+  } catch (error) {
+    throw error;
+  }
+});
+
+// /api/meals	POST	Adds a new meal to the database
+
+router.post("/", async (request, response) => {
+  const newMeal = await knex("meal").insert({
+    title: request.body.title,
+    description: request.body.description,
+    location: request.body.location,
+    when: request.body.when,
+    max_reservations: request.body.max_reservations,
+    price: request.body.price,
+    created_date: request.body.created_date,
+  });
+
+  response.status(201).json({ message: "Created a new meal", id: newMeal[0] });
+});
+
+// /api/meals/:id	GET	Returns the meal by id
+router.get("/:id", async (request, response) => {
+  try {
+    const idAvailable = await checkIfIdAvailable(request.params.id);
+
+    if (!idAvailable) {
+      response.status(400).json({ error: "id not available" });
+      return;
+    }
+    const meal = await knex("meal")
+      .where({ "meal.id": request.params.id })
+      .limit(1);
+    response.json(meal);
+  } catch (error) {
+    throw error;
+  }
+});
+
+router.get("/:meal_id/reviews", async (request, response) => {
+  try {
+    const idAvailable = await checkIfIdAvailable(request.params.meal_id);
+    const reviewAvailable = await checkIfReviewAvailable(
+      request.params.meal_id
+    );
```

Comment:
We don't need to check if there are reviews for this meal. If there are none, we just return an empty array.

Reply:
ok removed